pkg/resource/trail/custom_update_api.go

```diff
@@ -0,0 +1,135 @@
+package trail
+
+import (
+	"context"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/cloudtrail"
+)
+
+// customUpdateTrail implements a custom logic for handling Trail
+// resource updates.
+func (rm *resourceManager) customUpdateTrail(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (*resource, error) {
+	var err error
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.customUpdateTrail")
+	defer exit(err)
+
+	var updated *resource
+	updated = desired
```

Comment:
Did you mean to deepcopy this?

Reply:
Just a brave copy pasta from old ecr code